syntax = "proto3";

package core;
option go_package = "github.com/jalexanderII/zero_fintech/gen/Go/core";

import "common/common.proto";

message User {
  string id = 1;
  string username = 2;
  string email = 3;
  string password = 4;
}

message GetUserRequest {
  string id = 1;
}

message UpdateUserRequest {
  string id = 1;
  User user = 2;
}

message DeleteUserRequest {
  string id = 1;
}

message ListUserRequest {}

message ListUserResponse {
  repeated User users = 1;
}

message DeleteUserResponse {
  common.DELETE_STATUS status = 1;
  User user = 2;
}
